### PR TITLE
[SYCL] Collect sycl_declared_aspects metadata

### PR DESCRIPTION
**Pull request by KornevNikita**
In scope of optional kernel features implementation we should also collect sycl_declared_aspects metadata to create right "aspects" property
spec: https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md#create-the-sycldevice-requirements-property-set

### CODE REVIEW

**Comment by KornevNikita** [comment]
@intel/sycl-language-enabling-triage take a look please

File: llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp [hunk 1]
```diff
@@ -18,32 +18,29 @@ using namespace llvm;
...
+            cast<ConstantAsMetadata>(MDN->getOperand(I).get())->getValue();
+        Aspects.insert(
+            static_cast<uint32_t>(C->getUniqueInteger().getZExtValue()));
+      }
   };
 
-  // { LLVM-IR metadata name , [SYCL/Device requirements] property name }, see:
-  // https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md#create-the-sycldevice-requirements-property-set
-  // Scan the module and if the metadata is present fill the corresponing
-  // property with metadata's aspects
```
Comment (steffenlarsen) [inline llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp:31]: I feel like we are losing information by removing this. Would it make sense to keep this comment, potentially rephrased?

Comment (steffenlarsen) [reply llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp:31]: I agree, it's mostly the second part that still seems relevant.

Comment (KornevNikita) [reply llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp:31]: Done in 22fd515

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by KornevNikita** [comment]
Closing this PR because this work should be done in SYCLPropagateAspectsUsagePass.

**Comment by KornevNikita** [comment]
Another decision: https://github.com/intel/llvm/pull/7419

**Pull request closed without merging**